cloud-service-providers/aws/sagemaker/README_shell.md

```diff
@@ -0,0 +1,204 @@
+  * [Invocation](#invocation)
+- [Testing (Local)](#testing--local-)
+  * [Health](#health)
+  * [Invocation](#invocation-1)
+    + [Non-streaming](#non-streaming)
+    + [Streaming](#streaming)
+- [Cleanup](#cleanup)
+
+## Preparation
+
+> 🛈 **[Prefer python? Check here](README_python.md)**
+
+Customize the environment variables below to match your AWS, NGC, etc. configuration(s). If needed, customize the the parameters passed to the `launch.sh` call to ensure proper mapping of frontend/backend ports and source entrypoint. At a minimum you should customize the following:
+- `NGC_API_KEY`
+- `DST_REGISTRY`
+- `SG_INST_TYPE`
+  - Note that `ml.p4d.24xlarge` or similar variants are required for llama3-70b. `ml.g5.4xlarge` will work fine for `llama3-8b`
+- `SG_EXEC_ROLE_ARN`
+
+```bash
+git clone https://github.com/liveaverage/nim-shim && cd nim-shim
+
+### Set your NGC API Key
+export NGC_API_KEY=nvapi-your-api-key
+
+export SRC_IMAGE_PATH=nvcr.io/nim/meta/llama3-70b-instruct:latest
+export SRC_IMAGE_NAME="${SRC_IMAGE_PATH##*/}"
+export SRC_IMAGE_NAME="${SRC_IMAGE_NAME%%:*}"
+export DST_REGISTRY=your-registry.dkr.ecr.us-west-2.amazonaws.com/nim-shim
+
+docker login nvcr.io
+docker login ${DST_REGISTRY}
+docker pull ${SRC_IMAGE}
+
+# Build shimmed image
+sed 's/{{ SRC_IMAGE }}/$SRC_IMAGE/g' Dockerfile > Dockerfile.tmp
+envsubst < Dockerfile.tmp > Dockerfile.nim
+docker build -f Dockerfile.nim -t ${DST_REGISTRY}:${SRC_IMAGE_NAME} -t nim-shim:latest .
+docker push ${DST_REGISTRY}:${SRC_IMAGE_NAME}
+
+export SG_EP_NAME="nim-llm-${SRC_IMAGE_NAME}"
+export SG_EP_CONTAINER=${DST_REGISTRY}:${SRC_IMAGE_NAME}
+export SG_INST_TYPE=ml.p4d.24xlarge # ml.g5.4xlarge -- adequate for llama3-8b
+export SG_EXEC_ROLE_ARN="arn:aws:iam::YOUR-ARN-ROLE:role/service-role/AmazonSageMakerServiceCatalogProductsUseRole"
+export SG_CONTAINER_STARTUP_TIMEOUT=850 #in seconds -- adjust depending on dynamic or S3 model pull; model parameters (70b can take 460s+ to download)
+```
+
+## Usage
+
+### Deploying to Sagemaker
+
+Review logs in Cloudwatch. Ensure proper instance types have been set for the correlated model you're running & startup timeout values have been set to sane values, especially for dynamic download of large models (70b+).
+
+```bash
+# Generate model JSON
+envsubst < templates/sg-model.template > sg-model.json
+
+# Create Model
+aws sagemaker create-model \
+    --cli-input-json file://sg-model.json
+
+# Create Endpoint Config
+aws sagemaker create-endpoint-config \
+    --endpoint-config-name $SG_EP_NAME \
+    --production-variants "$(envsubst < templates/sg-prod-variant.template)"
+
+# Create Endpoint
+aws sagemaker create-endpoint \
+    --endpoint-name $SG_EP_NAME \
+    --endpoint-config-name $SG_EP_NAME
+```
+
+### Deploying locally
+
+Start the container and monitor for:
+- Caddy download & launch
+- Model weight(s) download
+- Service startup(s)
+
+
+```bash
+# Optional (but recommended to expedite future NIM launch times)
+mkdir -p /opt/nim/cache
```

Comment:
Should we make this `/opt/nim/.cache` to stay consistent? Personally I don't think it is necessary, just pointing it out.